pkg/pool-linear/contracts/aave/AaveLinearPoolFactory.sol

```diff
@@ -135,7 +165,24 @@ contract AaveLinearPoolFactory is
         address actualRebalancerAddress = Create2.deploy(0, rebalancerSalt, rebalancerCreationCode);
         require(expectedRebalancerAddress == actualRebalancerAddress, "Rebalancer deployment failed");
 
+        // Identify the protocolId associated with this pool. We do not require that the protocolId be registered.
+        emit AaveLinearPoolCreated(address(pool), protocolId);
+
         // We don't return the Rebalancer's address, but that can be queried in the Vault by calling `getPoolTokenInfo`.
         return pool;
     }
+
+    /**
+     * @notice Register an id (and name) to differentiate between multiple protocols using this factory.
+     * @dev This is a permissioned function. Protocol ids cannot be deregistered.
+     */
+    function registerProtocolId(uint256 protocolId, string memory name) external authenticate {
+        require(!_protocolIds[protocolId].registered, "Protocol already registered");
+
+        _registerProtocolId(protocolId, name);
+    }
+
+    function _registerProtocolId(uint256 protocolId, string memory name) private {
+        _protocolIds[protocolId] = ProtocolIdData({ name: name, registered: true });
```

Comment:
do we need the `registered` member if protocol IDs cannot be deregistered?

Reply:
We could read the stored value, hash it, and check against the hash of the empty string afaik. This isn't gas sensitive though so better to go for clarity.